packages/core/src/store/level-db-store.ts

```diff
@@ -61,23 +60,33 @@ class LevelDBStoreMap {
     return new Promise((res) => setTimeout(res, 100))
   }
 
-  private getDefaultLocation(): string {
-    return this.networkName === Networks.MAINNET ? OLD_ELP_DEFAULT_LOCATION : this.networkName
+  private getStoreLocation(useCaseName: string, networkName = this.networkName): string {
+    return useCaseName === DEFAULT_LEVELDB_STORE_USE_CASE_NAME
+      ? networkName
+      : `${networkName}-${useCaseName}`
   }
 
   private getFullLocation(useCaseName = DEFAULT_LEVELDB_STORE_USE_CASE_NAME): string {
-    if (useCaseName === DEFAULT_LEVELDB_STORE_USE_CASE_NAME) {
-      return this.getDefaultLocation()
-    } else {
-      return `${this.getDefaultLocation()}-${useCaseName}`
+    // Check if store exists at legacy ELP location
+    if (this.networkName === Networks.MAINNET) {
+      const elpLocation = this.getStoreLocation(useCaseName, OLD_ELP_DEFAULT_LOCATION)
+      const storePath = path.join(this.#storeRoot, elpLocation)
+      if (fs.existsSync(storePath)) {
```

Comment:
I believe as written this will wind up running this check every time the leveldb state store is accessed, which will both be inefficient and flood the logs with the warning message.  Can we do this check just once at startup and remember the results?